Tests/SonarScanner.MSBuild.Common.Test/ProcessRunnerTests.cs

```diff
@@ -300,6 +300,56 @@ public void ProcRunner_ArgumentQuotingForwardedByBatchScript()
             AssertExpectedLogContents(testDir, expected);
         }
 
+        [TestMethod]
+        [WorkItem(1706)] // https://github.com/SonarSource/sonar-scanner-msbuild/issues/1706
+        public void ProcRunner_ArgumentQuotingScanner()
+        {
+            // Checks arguments passed to a batch script which itself passes them on are correctly escaped
+            var testDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var batchName = TestUtils.WriteBatchFileForTest(TestContext,
+@"@echo off
+REM The sonar-scanner.bat uses %* to pass the argument to javac.exe
+echo %*
+REM Because of the escaping, the single arguments are somewhat broken on echo. A workaround is to add some new lines for some reason. 
+echo %1
+
+
+echo %2
+
+
+echo %3
+
+
+echo %4
+
+
+");
+            var logger = new TestLogger();
+            var runner = new ProcessRunner(logger);
+            var expected = new[]
+            {
+                @"-Dsonar.scanAllFiles=true",
+                @"-Dproject.settings=D:\DevLibTest\ClassLibraryTest.sonarqube\out\sonar-project.properties",
+                @"--from=ScannerMSBuild/5.13.1",
+                @"--debug"
+            };
+            var args = new ProcessRunnerArguments(batchName, true) { CmdLineArgs = expected, WorkingDirectory = testDir };
+            var success = runner.Execute(args);
```

Comment:
What is actually passed to the Test.bat looks like so:
```
^^^"^^^-^^^D^^^s^^^o^^^n^^^a^^^r^^^.^^^s^^^c^^^a^^^n^^^A^^^l^^^l^^^F^^^i^^^l^^^e^^^s^^^=^^^t^^^r^^^u^^^e^^^"^^^ ^^^"^^^-^^^D^^^p^^^r^^^o^^^j^^^e^^^c^^^t^^^.^^^s^^^e^^^t^^^t^^^i^^^n^^^g^^^s^^^=^^^D^^^:^^^\^^^D^^^e^^^v^^^L^^^i^^^b^^^T^^^e^^^s^^^t^^^\^^^C^^^l^^^a^^^s^^^s^^^L^^^i^^^b^^^r^^^a^^^r^^^y^^^T^^^e^^^s^^^t^^^.^^^s^^^o^^^n^^^a^^^r^^^q^^^u^^^b^^^e^^^\^^^o^^^u^^^t^^^\^^^s^^^o^^^n^^^a^^^r^^^-^^^p^^^r^^^o^^^j^^^e^^^c^^^t^^^.^^^p^^^r^^^o^^^p^^^e^^^r^^^t^^^i^^^e^^^s^^^"^^^ ^^^"^^^-^^^-^^^f^^^r^^^o^^^m^^^=^^^S^^^c^^^a^^^n^^^n^^^e^^^r^^^M^^^S^^^B^^^u^^^i^^^l^^^d^^^/^^^5^^^.^^^1^^^3^^^.^^^1^^^"^^^ ^^^"^^^-^^^-^^^d^^^e^^^b^^^u^^^g^^^"
```
Every argument is surrounded by double quotes and every single character is escaped by prepending `^^^`. This works somewhat but seems to break under certain conditions. The log from the issue indicates a Portuguese local which may have an impact too.